File: LeetCode/1457. Pseudo-Palindromic Paths in a Binary Tree/Solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int PseudoPalindromicPaths (TreeNode root) {
        if(root == null) return 0;

        var st = new Stack<(TreeNode, int)>();
        st.Push((root, 1 << root.val));
        int ans = 0;

        while(st.Count() != 0){
            var cur = st.Pop();

            bool noChild = true;
            if(cur.Item1.left != null) {
                noChild = false;
                st.Push((cur.Item1.left, cur.Item2 ^ (1 << cur.Item1.left.val)));
            }

            if(cur.Item1.right != null) {
                noChild = false;
                st.Push((cur.Item1.right, cur.Item2 ^ (1 << cur.Item1.right.val)));
            }

            if(noChild && (cur.Item2 & (cur.Item2 - 1)) == 0){
                ans++;
            }
        }

        return ans;
    }
}

File: LeetCode/1457. Pseudo-Palindromic Paths in a Binary Tree/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var pseudoPalindromicPaths = function (root) {
  let ans = 0;

  let stack = [[root, 0]];

  while (stack.length > 0) {
    let [node, path] = stack.pop();

    if (node !== null) {
      path ^= 1 << node.val;

      if (node.left === null && node.right === null) {
        ans += (path & (path - 1)) === 0 ? 1 : 0;
      } else {
        stack.push([node.left, path]);
        stack.push([node.right, path]);
      }
    }
  }

  return ans;
};
